std/range/package.d

```diff
@@ -4173,10 +4195,10 @@ private string lockstepMixin(Ranges...)(bool withIndex)
             }
             return res;
         }
-    }, params.join(", "), withIndex ? "size_t index = 0;" : "",
+    }, name, params.join(", "), indexDef,
        emptyChecks.join(" && "), dgArgs.join(", "),
        popFronts.join("\n                "),
-       withIndex ? "index++;" : "");
```

Comment:
The indexing behavior could be confusing, as `foreach_reverse` on an array counts the indices in reverse.
I wonder if its best to leave out the indexed overload for reverse, as we may not have a `length`, and when we do, what if the `length`s are different?